MetadataExtractor/IO/BufferReader.Indexed.cs

```diff
@@ -193,17 +195,11 @@ namespace MetadataExtractor.IO;
         }
         else
```

Comment:
Do we need the else block any more? Looks like the size check could be moved into a ternary conditional and the two blocks collapsed into one, as you've done elsewhere in the PR.

Reply:
unified the logic across all the readers. much tidier.